src/jdk.incubator.vector/share/classes/jdk/incubator/vector/AbstractMask.java

```diff
@@ -63,23 +63,11 @@
 
     @Override
     public boolean laneIsSet(int i) {
-        return getBits()[i];
-    }
-
-    @Override
-    public long toLong() {
-        // FIXME: This should be an intrinsic.
-        if (length() > Long.SIZE) {
-            throw new UnsupportedOperationException("too many lanes for one long");
-        }
-        long res = 0;
-        long set = 1;
-        boolean[] bits = getBits();
-        for (int i = 0; i < bits.length; i++) {
-            res = bits[i] ? res | set : res;
-            set = set << 1;
+        if (length() <= Long.SIZE) {
```

Comment:
We need to check that `i > 0 && i < length()` and if not throw `IllegalArgumentException`.

Currently the behavior is unspecified, can you please add the following to the documentation of `VectorMask.laneIsSet`. (I will handle any CSR related tasks.)

```
     * @throws IllegalArgumentException if the index is is out of range
     * ({@code < 0 || >= length()})
```